### PR TITLE
Add prismic article parser

### PR DESCRIPTION
**Pull request by jamesgorrie**
References #981 

## Type

✨ Feature  

## Value
We can start using another CMS using our new model, meaning we can start taking over other bits of content on the site, namely events, exhibitions etc.

If you look at how we create an article here, this would replace all the `ArticleFactory.fromWpApi` stuff and `bodyParser` crap.

## TODO:
- [x] ~Use image resizing tool~ (we can do this after the editors have had a go)
- [x] ~Preview~ (we can publish to live as these pages aren't listed anywhere, so no problem for testing)
- [x] ~Instagram embeds aren't using the JS for some reason~ (we don't use them that much so can figure this out as we go on)
- [x] ~The multitude of tech TODOs in the file~ (a refactoring for when we kill wordpress).
- [x] Author

## Screenshot
prismicrednder

### All
- [x] Demoed to the relevant people
- [x] PR labelled and assigned
- [x] Any introduced code complexity has been flagged

### CODE REVIEW

**Comment by jamesgorrie** [comment]
@jennpb FYI ⬆️

File: server/routes/index.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ r.get('/', index);
 r.get('/healthcheck', healthcheck);
 r.get('/explore', explore);
 r.get('/articles', articles);
+r.get('/articles/(W):id', prismicArticle);
```
Comment (davidpmccormick) [inline server/routes/index.js]: Does this mean 'the-`id`-is-whatever-this-string-is-without-a-`W`-at-the-front-if-there-is-one'?

File: server/services/prismic-content.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+  const bodyParts = prismicArticle.data.body.map(slice => {
+    switch (slice.slice_type) {
+      case 'standfirst':
+        return {
+          type: 'standfirst',
+          weight: 'default',
+          value: RichText.asHtml(slice.primary.text)
+        };
+        break;
```
Comment (davidpmccormick) [inline server/services/prismic-content.js]: I think these `break`s would be linting errors as they're unreachable after the `return`s

File: server/routes/index.js (hunk 1 above)
Comment (jamesgorrie) [inline server/routes/index.js]: We can safely assume prismic IDs start with `W`.
I am going to assume no Wordpress article do.

So if we hit a `W` => Prismic, else Wordpress.

File: server/services/prismic-content.js (hunk 2 above)
Comment (jamesgorrie) [inline server/services/prismic-content.js]: Seems my linter is out of whack again.

File: server/routes/index.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import Router from 'koa-router';
-import {index, article, articles, explore, healthcheck, featureFlags, performanceTest, explosion, preview, series} from '../controllers';
+import {index, article, articles, explore, healthcheck, featureFlags, performanceTest, explosion, preview, series, prismicArticle} from '../controllers';
```
Comment (jakubfiala) [inline server/routes/index.js]: for readability, could you break the imports list into separate lines?
```js
import {
  index,
  article,
  articles,
  explore,
  healthcheck,
  featureFlags,
  performanceTest,
  explosion,
  preview,
  series,
  prismicArticle
} from '../controllers';
```

File: server/services/prismic-content.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  // TODO: potentially get rid of this
+  const publishDate = PrismicDate(prismicArticle.data.publishDate || prismicArticle.first_publication_date);
+
+  // TODO:
+  const mainMedia = prismicArticle.data.body.filter(slice => slice.primary.weight === 'featured').map(slice => {
+    return prismicImageToPicture(slice.primary);
+  });
+
+  // TODO: Don't convert this into thumbnail
+  const promo = prismicArticle.data.promo.find(slice => slice.slice_type === 'embeddedImage').primary;
```
Comment (jakubfiala) [inline server/services/prismic-content.js]: what if `find` returns `undefined`?

File: server/services/prismic-content.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+  const series = [] || prismicArticle.data.series.length > 0 && prismicArticle.data.series.map(prismicSeries => {
+      const seriesData = prismicSeries.primary.series.data;
+      return {
+        name: seriesData.name,
+        description: seriesData.description,
+        commissionedLength: seriesData.commissionedLength,
+        positionInSeries: prismicSeries.primary.positionInSeries
+      };
+    });
```
Comment (jakubfiala) [inline server/services/prismic-content.js]: What does this do? Doesn't `[] || anything` always return `[]` and not evaluate `anything`?

File: server/services/prismic-content.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    url: url,
+    datePublished: publishDate,
+    thumbnail: thumbnail,
+    author: author,
+    series: series,
+    bodyParts: bodyParts,
+    mainMedia: mainMedia
+  };
+
+  return article;server/controllers/index.js
```
Comment (jakubfiala) [inline server/services/prismic-content.js]: what's this syntax doing?

File: server/routes/index.js (hunk 3 above)
Comment (jamesgorrie) [inline server/routes/index.js]: If someone wants to put this in the lint - then yip.
I personalyl find 1 line easier to read, so long as it's not 100+ characters.

File: server/services/prismic-content.js (hunk 6 above)
Comment (jamesgorrie) [inline server/services/prismic-content.js]: 🐣  dang copy/paste error, thanks for spotting.

File: server/services/prismic-content.js (hunk 4 above)
Comment (jamesgorrie) [inline server/services/prismic-content.js]: Yea - there's loads of `null` pointer exceptions here.

A bit of context here - this is just so editors can start testing Prismic, so I can't say this is the best code I've written.

I'll 100% get it typed etc though once we're 100% commited to this being the system.

File: server/services/prismic-content.js (hunk 5 above)
Comment (jamesgorrie) [inline server/services/prismic-content.js]: That was for debuggin - thanks for catching.

File: server/services/prismic-content.js (hunk 2 above)
Comment (jamesgorrie) [inline server/services/prismic-content.js]: I'm going to leave it out of this PR - there's a bit of work needed around linting etc.

File: client/js/components/instagram.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 function init() {
-  const hasInstagramEmbed = Boolean(document.querySelector('.instagram-media'));
+  const hasInstagramEmbed = Boolean(document.querySelector('.instagram-media')) || Boolean(document.querySelector('[data-type="instapp:photo"]'));
+  console.info(hasInstagramEmbed);
```
Comment (davidpmccormick) [inline client/js/components/instagram.js]: do we need this?

**Pull request merged**